### DISCUSSION TITLE
Stepping through object init blocks

### DISCUSSION THREAD

**Discussion by TahirAhmadov** [dotnet/roslyn#63382] [Q&A] [answered]
Before I open a feature request with Visual Studio, I was wondering if C# compiler creates enough data in the PDB file to make it possible to step through object initializer blocks? I find that I need it more and more, given the powerful syntax for collections,etc., sometimes I end up compacting a lot of code into the init block, and currently it's only "one step" - no way to set a breakpoint in the middle, see where exactly an exception is raised from, etc.

**Comment by bernd5** [comment]
This is not a language question - but a debugger question.
From language perspective there is no "stepping through" and no pdb file...

**Comment by 333fred** [comment, marked as answer]
@TahirAhmadov please create a feature request on roslyn.

**Reply by TahirAhmadov** [reply]
Done, as "IDE feature" category.
https://github.com/dotnet/roslyn/issues/63383
